packages/core/chrome/core-chrome-browser-internal/src/ui/project/header.tsx

```diff
@@ -6,76 +6,153 @@
  * Side Public License, v 1.
  */
 
-import React, { createRef, useState } from 'react';
-import { Router } from 'react-router-dom';
 import {
   EuiHeader,
   EuiHeaderLogo,
   EuiHeaderSection,
   EuiHeaderSectionItem,
   EuiHeaderSectionItemButton,
   EuiIcon,
+  EuiLoadingSpinner,
   htmlIdGenerator,
 } from '@elastic/eui';
+import { css } from '@emotion/react';
+import type { InternalApplicationStart } from '@kbn/core-application-browser-internal';
 import {
   ChromeBreadcrumb,
   ChromeGlobalHelpExtensionMenuLink,
   ChromeHelpExtension,
   ChromeNavControl,
 } from '@kbn/core-chrome-browser/src';
-import useLocalStorage from 'react-use/lib/useLocalStorage';
-import { i18n } from '@kbn/i18n';
-import { Observable } from 'rxjs';
+import type { HttpStart } from '@kbn/core-http-browser';
 import { MountPoint } from '@kbn/core-mount-utils-browser';
-import { InternalApplicationStart } from '@kbn/core-application-browser-internal';
-import { HeaderBreadcrumbs } from '../header/header_breadcrumbs';
+import { i18n } from '@kbn/i18n';
+import React, { createRef, useState } from 'react';
+import { Router } from 'react-router-dom';
+import useLocalStorage from 'react-use/lib/useLocalStorage';
+import useObservable from 'react-use/lib/useObservable';
+import type { Observable } from 'rxjs';
 import { HeaderActionMenu, useHeaderActionMenuMounter } from '../header/header_action_menu';
+import { HeaderBreadcrumbs } from '../header/header_breadcrumbs';
 import { HeaderHelpMenu } from '../header/header_help_menu';
 import { HeaderNavControls } from '../header/header_nav_controls';
 import { ProjectNavigation } from './navigation';
 
+const headerCss = {
+  logo: {
+    container: css`
+      display: inline-block;
+      min-width: 56px; /* 56 = 40 + 8 + 8 */
+      padding: 0 8px;
+      cursor: pointer;
+    `,
+    logo: css`
+      min-width: 0; /* overrides min-width: 40px */
+      padding: 0;
+    `,
+    spinner: css`
+      position: relative;
+      left: 4px;
+      top: 2px;
+    `,
+  },
+  nav: {
+    toggleNavButton: css`
+      border-right: 1px solid #d3dae6;
```

Comment:
Not sure if there is a cleaner way to do this in CSS. It may be better to leave the header style alone for now since changes will be coming to this area in an EUI update.

cc @cee-chen

Reply:
You'll need to use the `useEuiTheme` hook to get the JS theme variables for border width/colors. But yeah I'll be working on EuiCollapsibleNav updates next week here, which will include this styling, so if you want you could leave this out!

Reply:
@cee-chen thanks for the updates! I'm tempted to update this PR to use `useEuiTheme` and ensure the border for this header item (the nav show/hide button) is the right color.

This discussion probably belongs here: https://github.com/elastic/eui/issues/6759#issuecomment-1579779907